core/o11y/src/lib.rs

```diff
@@ -131,8 +141,8 @@ where
 
 // Constructs an OpenTelemetryConfig which sends span data to an external collector.
 // OpenTelemetryConfig is currently empty, but more configuration will be added to it later.
-fn make_opentelemetry_layer<S>(
-    config: &OpenTelemetryConfig,
+async fn make_opentelemetry_layer<S>(
```

Comment:
Why did it become async?

Reply:
This is not super obvious, but constructing a `Tokio` otel pipeline requires the implicit async context so that it can set up its channels, timers and whatnot. `default_context` has been made `async` for that reason, but I felt that making `make_opentelemetry_layer` async makes it clearer as to where exactly the dependency on `async` comes in `default_context` in the first place.

I should also write a comment…